tools/Importer/Importer.cs

```diff
@@ -0,0 +1,341 @@
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Text;
+using System.Threading;
+using Azure.Storage.Blobs;
+using Azure.Storage.Blobs.Models;
+using Microsoft.Health.Fhir.Store.Utils;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Health.Fhir.Importer
+{
+    internal static class Importer
+    {
+        private static readonly HttpClient HttpClient = new HttpClient();
+        private static readonly string Endpoints = ConfigurationManager.AppSettings["FhirEndpoints"];
+        private static readonly string ConnectionString = ConfigurationManager.AppSettings["ConnectionString"];
+        private static readonly string ContainerName = ConfigurationManager.AppSettings["ContainerName"];
+        private static readonly int BlobRangeSize = int.Parse(ConfigurationManager.AppSettings["BlobRangeSize"]);
+        private static readonly int ReadThreads = int.Parse(ConfigurationManager.AppSettings["ReadThreads"]);
+        private static readonly int WriteThreads = int.Parse(ConfigurationManager.AppSettings["WriteThreads"]);
+        private static readonly int NumberOfBlobsToSkip = int.Parse(ConfigurationManager.AppSettings["NumberOfBlobsToSkip"]);
+        private static readonly int MaxBlobIndexForImport = int.Parse(ConfigurationManager.AppSettings["MaxBlobIndexForImport"]);
+        private static readonly int ReportingPeriodSec = int.Parse(ConfigurationManager.AppSettings["ReportingPeriodSec"]);
+        private static readonly int MaxRetries = int.Parse(ConfigurationManager.AppSettings["MaxRetries"]);
+        private static readonly bool UseStringJsonParser = bool.Parse(ConfigurationManager.AppSettings["UseStringJsonParser"]);
+        private static readonly bool UseStringJsonParserCompare = bool.Parse(ConfigurationManager.AppSettings["UseStringJsonParserCompare"]);
+
+        private static long totalReads = 0L;
+        private static long readers = 0L;
+        private static Stopwatch swReads = Stopwatch.StartNew();
+        private static long totalWrites = 0L;
+        private static long writers = 0L;
+        private static long epCalls = 0L;
+        private static long waits = 0L;
+        private static List<string> endpoints;
+
+        internal static void Run()
+        {
+            if (string.IsNullOrEmpty(Endpoints))
+            {
+                throw new ArgumentException("FhirEndpoints value is empty");
+            }
+
+            endpoints = Endpoints.Split(";", StringSplitOptions.RemoveEmptyEntries).ToList();
+
+            var globalPrefix = $"RequestedBlobRange=[{NumberOfBlobsToSkip + 1}-{MaxBlobIndexForImport}]";
+            Console.WriteLine($"{globalPrefix}: Starting...");
+            var blobContainerClient = GetContainer(ConnectionString, ContainerName);
+            var blobs = blobContainerClient.GetBlobs().OrderBy(_ => _.Name).Where(_ => _.Name.EndsWith(".ndjson", StringComparison.OrdinalIgnoreCase)).ToList();
```

Comment:
Would it be better to apply the Where before the OrderBy? i.e.:
blobContainerClient.GetBlobs().Where(_ => _.Name.EndsWith(".ndjson", StringComparison.OrdinalIgnoreCase)).OrderBy(_ => _.Name).ToList();

Reply:
Good point. In reality this difference is small. I don't want to risk extra failures in build, so suggest to leave as is.